server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -289,12 +306,25 @@ protected void closeInternal() {
                 }
             }
         };
+
+        final String nodeName = Objects.requireNonNull(Node.NODE_NAME_SETTING.get(settings));
+        danglingIndicesThreadPoolExecutor = EsExecutors.newScaling(
+            nodeName + "/" + DANGLING_INDICES_UPDATE_THREAD_NAME,
+            1, 1,
```

Comment:
If `gateway.write_dangling_indices_info` is `false` I think we should avoid creating a thread here.

Reply:
👍 done in 3a7b656f5ce